sim: or1k: Get building again

stffrdhrn opened the pull request:
The or1k sim here is was we want to push to upstream since it does not depend on or1ksim. 
I have enabled the sim build in travis since it should build now.

stffrdhrn committed:
sim: or1k: Get or1k sim building with latest sim common

There have been a few type name changes.
 * Update configure.ac to use new ENDIAN config tpes
 * Update mloop to use new CGEN_TRACE.. instead of TRACE..
 * Change singnature of sim_create_inferior to have const param
 * Remove sim_close as its covered by sim_close in common now
 * Move WITH_SCACHE_PBB into sim-main.h from tconfig.in
 * Remove SIM_HANDLES_LMA as it seems its not needed anymore from tcofig.in
 * Remove tconfig.in

stffrdhrn committed:
sim: or1k: Regenerate cgen files

These files have been regenerated using latest cgen which has been
updated to generate code that is compatible with the current sim
common api.

stffrdhrn committed:
sim: or1k: Regenerate autotool files

After chaning the configure.am this is require

stffrdhrn committed:
travis: update continuous build to build sim/or1k

Update the config line to enable sim build, or1k not or1ksim
The build depends on guile-1.8
On my machine I hacked the sim/common/Make-common.in and
opcodes/Makefile.am to specify guile as guile1.8. But seeing if this
is not needed.

stffrdhrn committed:
Merge branch 'or1k' of https://github.com/openrisc/binutils-gdb into or1k-sim

stffrdhrn committed:
travis: Doesnt need guile, also make config more clear

The current config line says --disable-or1ksim but its a bit unclear. We
are actually enabling sim (or1k) but disabiling the or1ksim wrapper.
Add a --enable-sim line to be a bit more clear that sim is enabled.
Also remove guile-1.8 dependency since thats only needed for generating
the cgen stuff which is already generated.

stffrdhrn committed:
sim: or1k: Implement register store/fetch

There was a comment questioning the use of these. But as soon
as I run any basic test gdb complains that these are not implemented.

More tests and basic run and execute work now.

stffrdhrn committed:
sim: or1k: Do trap breakpoint handling

Breakpoints were not implemented in the sim.  This change allows for
trap instructions (inserted when gdb sets breakpoints) to transfer
control back to GDB allowing breakpoints to work with 'target sim'

After implementing this gdb test suite is looking better with:

                === gdb Summary ===

 # of expected passes            15981
 # of unexpected failures        509
 # of expected failures          27
 # of known failures             47
 # of unresolved testcases       34
 # of untested testcases         174
 # of unsupported tests          309

stffrdhrn committed:
sim: or1k: Implement fetch/store for ppc and sr

This was causing some tests to failure due to warnings.

stffrdhrn committed:
gdb: testsuite: Add or1k l.nop inscruction

The test case requires adding a nop instruction.  For or1k the
instruction is `l.nop`. This change uses the correct operation.

stffrdhrn committed:
gdb: or1k: Update calling conventions to help pass tests

The calling conventions for handling struct/array return, large scalars
and variadic args were incomplete and causing several gdb testsuite
tests to fail.

These changes were developed with help from olofk.

imphil commented on gdb/or1k-tdep.c:
```diff
@@ -463,7 +464,7 @@ or1k_return_value (struct gdbarch  *gdbarch,
       return RETURN_VALUE_ABI_RETURNS_ADDRESS;
     }
-
+  
```
trailing whitespace

stffrdhrn replied:
thanks, fixed this and others

imphil commented on gdb/or1k-tdep.c:
```diff
@@ -439,12 +439,13 @@ or1k_return_value (struct gdbarch  *gdbarch,
-      (rv_size >  bpw))
+  /* Deal with struct/union as addresses. If an array won't fit in a single 
+     register it is returned as address. Anything larger than 2 registers needs
+     to also be passed as address (this is maily from default_return_in_memory
```
just a drive-by: s/maily/mainly/

stffrdhrn replied:
thanks fixed this.

stffrdhrn committed:
or1k: Fix whitespace issues and a typo

There were a few places where I had trailing whitespaces those have
been removed. The file should be clean now.

Also fixed a typo in a comment maily/mainly, I reworded to remove
completely, since mainly is a meaningless word.
